src/status_im/contexts/settings/wallet/saved_addresses/save_address/view.cljs

```diff
@@ -17,50 +17,48 @@
 
 (defn view
   []
-  (let [{:keys [edit?]} (rf/sub [:get-screen-params])
-        {:keys [address name customization-color ens ens?]}
-        (rf/sub [:wallet/saved-address])
+  (let [{:keys [address name customization-color ens
+                ens? edit?]}              (rf/sub [:get-screen-params])
         [address-label set-address-label] (rn/use-state (or name ""))
```

Comment:
This was a root of the problem - we were using data from the subscription and locking it inside the state. So if screen opened before the subscription value updated we ended up with wrong data saved in state hook. Solution is to pass the correct data as screen params instead of subs.